bigfastapi/notification.py

```diff
@@ -0,0 +1,155 @@
+import fastapi
+from fastapi import Request, APIRouter
+from datetime import datetime
+import sqlalchemy.orm as orm
+from bigfastapi import auth
+from bigfastapi.db.database import get_db
+from fastapi import Depends
+
+from bigfastapi.schemas import users_schemas
+from .models import notification_models
+from .schemas import notification_schemas
+from uuid import uuid4
+from typing import List
+from bigfastapi.auth import is_authenticated
+
+
+app = APIRouter(tags=["Notification"])
+
+@app.get("/notification/{notification_id}", response_model=notification_schemas.Notification)
+def get_a_notification(
+    notification_id: str, 
+    db: orm.Session = Depends(get_db)
+):
+    return notification_selector(id=notification_id, db=db)
+
+@app.get("/notifications", response_model=List[notification_schemas.Notification])
+def get_all_notifications(
+    db: orm.Session = Depends(get_db)
+):
+    return get_all_notifications_from_db(db=db)
+
+@app.post("/notification", response_model=notification_schemas.Notification)
+def create_notification(
+    notification: notification_schemas.NotificationCreate, 
+    user: users_schemas.User = Depends(is_authenticated),
+    db: orm.Session = Depends(get_db)
+):
+    return create_a_notification(user=user, notification=notification, db=db)
+
+@app.put("/notification/{notification_id}/read", response_model=notification_schemas.Notification)
+def mark_notification_read(
+    notification_id: str,
+    db: orm.Session = Depends(get_db)
+):
+    return mark_a_notification_read(id=notification_id, db=db)
+
+@app.put("/notifications/read", response_model=List[notification_schemas.Notification])
+def mark_notifications_read(
+    db: orm.Session = Depends(get_db)
+):
+    return mark_all_notifications_read(db=db)
+
+@app.put("/notifications/{notification_id}", response_model=notification_schemas.Notification)
+def update_notification(
+    notification_id: str,
+    notification: notification_schemas.NotificationUpdate,
+    db: orm.Session = Depends(get_db)
+):
+    return update_a_notification(id=notification_id, notification=notification, db=db)
+
+@app.delete("/notification/{notification_id}")
+def delete_notification(
+    notification_id: str,
+    db: orm.Session = Depends(get_db)
+):
+    return delete_a_notification(id=notification_id, db=db)
+
+
+
+
+#=================================== NOTIFICATION SERVICES =================================#
+
+def get_authenticated_user_email(user: users_schemas.User):
+    return user.email
+
+def notification_selector(id: str, db: orm.Session):
+    notification = db.query(notification_models.Notification).filter(notification_models.Notification.id == id).first()
+
+    if notification is None:
+        raise fastapi.HTTPException(status_code=404, detail="Notification does not exist")
```

Comment:
The entire point of services is to abstract the HTTP thing away

Reply:
I don't seem to understand what you mean here